opal/mca/accelerator/cuda/accelerator_cuda.c

```diff
@@ -435,34 +481,23 @@ static int accelerator_cuda_memcpy(int dest_dev_id, int src_dev_id, void *dest,
         return OPAL_SUCCESS;
     }
 
-    /* Async copy then synchronize is the default behavior as some applications
-     * cannot utilize synchronous copies. In addition, host memory does not need
-     * to be page-locked if an Async memory copy is done (It just makes it synchronous
-     * which is what we want anyway):
-     * https://docs.nvidia.com/cuda/cuda-c-programming-guide/index.html#concurrent-execution-host-device
-     * Additionally, cuMemcpy is not necessarily always synchronous. See:
-     * https://docs.nvidia.com/cuda/cuda-driver-api/api-sync-behavior.html
-     * TODO: Add optimizations for type field */
-    result = cuMemcpyAsync((CUdeviceptr) dest, (CUdeviceptr) src, size, opal_accelerator_cuda_memcpy_stream);
-    if (OPAL_UNLIKELY(CUDA_SUCCESS != result)) {
-        opal_show_help("help-accelerator-cuda.txt", "cuMemcpyAsync failed", true, dest, src,
-                       size, result);
-        return OPAL_ERROR;
-    }
-    result = cuStreamSynchronize(opal_accelerator_cuda_memcpy_stream);
+    result = cuMemcpy((CUdeviceptr) dest, (CUdeviceptr) src, size);
     if (OPAL_UNLIKELY(CUDA_SUCCESS != result)) {
-        opal_show_help("help-accelerator-cuda.txt", "cuStreamSynchronize failed", true,
+        opal_show_help("help-accelerator-cuda.txt", "cuMemcpy failed", true,
                        OPAL_PROC_MY_HOSTNAME, result);
         return OPAL_ERROR;
     }
     return OPAL_SUCCESS;
 }
 
-static int accelerator_cuda_memmove(int dest_dev_id, int src_dev_id, void *dest, const void *src, size_t size,
-                             opal_accelerator_transfer_type_t type)
+static int accelerator_cuda_memmove_async(int dest_dev_id, int src_dev_id, void *dest,
```

Comment:
Indeed, we allocate memory to handle overlap. I guess there could be a better implementation, we just haven't looked for one because we didn't need it (the function was there before, we just added the async version)